### PR TITLE
Add support for additional `apoc.convert` functions

### PR DESCRIPTION
**Pull request by tonijurjevic96**
### Description

Please briefly explain the changes you made here.

### Pull request type

- [ ] Bugfix
- [x] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [x] Core algorithm/module implementation
- [x] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [x] Tests provided (unit / e2e)
- [ ] Code documentation
- [ ] README short description

### Documentation checklist
- [ ] Add the documentation label tag
- [ ] Add the bug / feature label tag
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[Release note text]**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [ ] Tag someone from docs team in the comments

## Summary by CodeRabbit

- **New Features**
  - Added a procedure to recursively flatten nested lists into a single-level list.
  - Introduced a procedure to convert Memgraph graph elements (nodes, edges, paths) into nested tree-like dictionary structures, with support for property filtering.
  - Added utilities for converting Memgraph and Python types to JSON strings and parsing JSON lists, including support for datetime and timedelta types.
  - Provided procedures to load JSON from local files and remote URLs into Memgraph.
- **Bug Fixes**
  - None.
- **Tests**
  - Added end-to-end and unit tests for list flattening, graph-to-tree conversion, and JSON utilities, covering a wide range of scenarios and edge cases.
  - Introduced mock modules and fixtures to support isolated testing of graph and JSON utilities.
- **Chores**
  - Updated subproject references and added module-level docstrings for clarity.

### CODE REVIEW

**Comment by Josipmrden** [comment]
Deadline to merge: 9th July

File: cpp/collections_module/algorithm/collections.cpp [hunk 1]
```diff
@@ -540,3 +540,63 @@ void Collections::Partition(mgp_list *args, mgp_graph *memgraph_graph, mgp_resul
...
+
+// Helper function to recursively flatten a list
+void FlattenHelper(const mgp::Value &value, mgp::List &result) {
+    if (value.IsNull()) {
+        // Skip null values
+        return;
+    }
+
+    if (value.IsList()) {
+        const auto &list = value.ValueList();
```
Comment (DavIvek) [inline cpp/collections_module/algorithm/collections.cpp:554]: I think that you don't need ref here

File: cpp/collections_module/algorithm/collections.cpp [hunk 2]
```diff
@@ -540,3 +540,63 @@ void Collections::Partition(mgp_list *args, mgp_graph *memgraph_graph, mgp_resul
...
 }
+
+namespace {
+
+// Helper function to recursively flatten a list
+void FlattenHelper(const mgp::Value &value, mgp::List &result) {
+    if (value.IsNull()) {
+        // Skip null values
+        return;
+    }
```
Comment (DavIvek) [inline cpp/collections_module/algorithm/collections.cpp:551]: This check isn't needed.

**Review by andrejtonev — APPROVED** [review]
